integrationTests/singleShard/block/executingMiniblocksSc_test.go

```diff
@@ -148,57 +232,396 @@ func TestProcessesJoinGameOf100PlayersRewardAndEndgame(t *testing.T) {
 	}
 
 	fmt.Println("Delaying for nodes p2p bootstrap...")
-	time.Sleep(stepDelay)
+	time.Sleep(time.Second)
 
 	round := uint64(0)
 	round = incrementAndPrintRound(round)
 
-	idxProposer := 0
 	hardCodedSk, _ := hex.DecodeString("5561d28b0d89fa425bbbf9e49a018b5d1e4a462c03d2efce60faf9ddece2af06")
 	hardCodedScResultingAddress, _ := hex.DecodeString("000000000000000000005fed9c659422cd8429ce92f8973bba2a9fb51e0eb3a1")
 	nodes[idxProposer].LoadTxSignSkBytes(hardCodedSk)
 
 	initialVal := big.NewInt(10000000)
 	topUpValue := big.NewInt(500)
-	withdrawValue := big.NewInt(10)
 	integrationTests.MintAllNodes(nodes, initialVal)
+	integrationTests.MintAllPlayers(nodes, players, initialVal)
 
 	deployScTx(nodes, idxProposer, string(scCode))
-	proposeBlock(nodes, idxProposer, round)
-	syncBlock(t, nodes, idxProposer, round)
+	time.Sleep(time.Second)
+	proposeBlock(nodes, []int{idxProposer}, round)
+	time.Sleep(time.Second)
+	syncBlock(t, nodes, []int{idxProposer}, round)
 	round = incrementAndPrintRound(round)
 
-	rMonitor := &statistics.ResourceMonitor{}
-	fmt.Println(rMonitor.GenerateStatistics())
+	runMultipleRoundsOfTheGame(
+		t,
+		100,
+		nrPlayers,
+		nodes,
+		players,
+		topUpValue,
+		hardCodedScResultingAddress,
+		round,
+		maxShards,
+		[]int{idxProposer},
+	)
+
+	checkRootHashes(t, nodes, []int{0})
+
+	time.Sleep(time.Second)
+}
+
+func TestProcessesJoinGame100PlayersMultipleTimesRewardAndEndgameInMultipleRoundsMultiShard(t *testing.T) {
```

Comment:
This should be moved out of this package: this is singleShard

Reply:
yeah, but currently a lot of functions are only in this package.

Reply:
Beni already move them in integrationTests package

Reply:
not all

Reply:
But should be split. It doesn't make sense to have in a subpackage called singleShard 2 multishard tests.